### ISSUE TITLE
Prevent certain long post from collapsing

### ISSUE THREAD

**Issue by harshilsharma2** [mattermost/mattermost-server#9843]
Currently in Mattermost a post collapses if it's longer than a certain height. Is there any `prop` in `Post.Props` that can be set to prevent this?

#### Context

I'm creating some sort of summarized reports from a plugin and they gets collapsed all the time. Due to the nature of this post, it offers a much better UX if it's not collapsed, preventing the user from expanding all the posts in the channel and even preventing them from missing on some information.

I'm aware that I can create a custom post type for this but I think it would be overkill for this basic task.

I'd be more than happy to submit a PR about this if you think it's something that can be a feature of Mattermost.

Thanks!

**Comment by amyblais** [comment]
Hi @harshilsharma63,

Currently there is no way to disable the "Show More" button feature, but an option would be to add a setting for that.

If you have forked Mattermost, you would just need to change these two values to something very high to stop it from collapsing
https://github.com/mattermost/mattermost-webapp/blob/master/components/post_view/post_message_view/post_message_view.jsx#L15
https://github.com/mattermost/mattermost-webapp/blob/master/sass/layout/_post.scss#L1898.

If you haven't forked Mattermost, I'm sure a PR to add a setting to optionally disable the "Show More" button will be accepted. Does this help?

**Comment by harshilsharma63** [comment]
@amyblais I'd love to submit a PR for this. I was thinking of adding a prop in `Post.Props` to prevent from collapsing the post. Does it sound right?

**Comment by amyblais** [comment]
Hi @harshilsharma63, letting you know that we're discussing this as a team and we're thinking that maybe an actual setting to disable collapsing long posts might be even a better option. We've gotten some feedback from other people as well that it might be better to just have long posts un-collapsed

**Comment by harshilsharma63** [comment]
@amyblais I think it makes sense. Are we looking at a system-wide setting or a user preference?

**Comment by amyblais** [comment]
Hi @harshilsharma63, I opened a ticket here for our UX team to first consider internally what type of setting it would be: https://mattermost.atlassian.net/browse/MM-13149. So far there was discussion that it might be a user-specific setting.

I made a note in the ticket that you are interested in helping implement this improvement, so we can then assign this ticket to you if you're still interested once a final decision / plan is done.

**Comment by harshilsharma63** [comment]
@amyblais awesome!

**Comment by harshilsharma63** [comment]
@amyblais any update on this?

**Comment by amyblais** [comment]
Hi @harshilsharma63, not yet, one of our product managers added a comment on the ticket that we need to do competitor research, consider client vs server settings, etc. before this ticket can be worked on. I'll ask the PM whether this is something that his developer team can prioritize.

**Comment by harshilsharma63** [comment]
@amyblais no worries!

**Comment by numericillustration** [comment]
After upgrading from 4.9 to 5.14 we've noticed this behavior change too where previously posts up to the 4000 char limit would show the whole post without a "show more" button.  While it's neat that we can post longer messages now both as users and via integrations, a user configurable setting for the hiding of the rest with a "show more" button would be great.

**Comment by numericillustration** [comment]
> Hi @harshilsharma63,
> 
> Currently there is no way to disable the "Show More" button feature, but an option would be to add a setting for that.
> 
> If you have forked Mattermost, you would just need to change these two values to something very high to stop it from collapsing
> https://github.com/mattermost/mattermost-webapp/blob/master/components/post_view/post_message_view/post_message_view.jsx#L15
> https://github.com/mattermost/mattermost-webapp/blob/master/sass/layout/_post.scss#L1898.
> 
> If you haven't forked Mattermost, I'm sure a PR to add a setting to optionally disable the "Show More" button will be accepted. Does this help?

looks like the code places for adjusting this have all changed with https://github.com/mattermost/mattermost-webapp/pull/1520/files but perhaps it would be easy to set a variable or something in place of everywhere its set to 600 (pixels?) and let users override that in their settings or a global override in the config.json file

**Comment by tmdoit** [comment]
I use Mattermost heavily for slash commands, there are such commands that I want to prevent result to being collapsed because it forces an extra click which is annoying after some time. There are of course use cases for collapse behaviour, possibility to set it via payload (POST or GET) could be a good way to go?

**Comment by amyblais** [comment]
The Jira ticket was closed as "won't fix" - if you're still interested in this, please help open a feature request here https://mattermost.uservoice.com/forums/306457-general!

**Comment by localparty** [comment]
this "feature" adds friction to read content, which is the most important use of the app
please make this top priority
